packages/phone-number-privacy/signer/src/domain/domain.service.ts

```diff
@@ -69,17 +48,27 @@ export class DomainService implements IDomainService {
     request: Request<{}, {}, unknown>,
     response: Response<DisableDomainResponse>
   ): Promise<void> {
-    Counters.requests.labels(DomainEndpoint.DISABLE_DOMAIN).inc()
+    const logger = response.locals.logger
+    const endpoint = DomainEndpoint.DISABLE_DOMAIN
+    Counters.requests.labels(endpoint).inc()
 
     // Check that the body contains the correct request type.
     if (!disableDomainRequestSchema(DomainSchema).is(request.body)) {
-      respondWithError(DomainEndpoint.DISABLE_DOMAIN, response, 400, WarningMessage.INVALID_INPUT)
-      return
+      return this.sendFailureResponse(response, WarningMessage.INVALID_INPUT, 400, endpoint, logger)
+    }
+
+    if (!config.api.domains.enabled) {
+      return this.sendFailureResponse(
+        response,
+        WarningMessage.API_UNAVAILABLE,
+        501,
```

Comment:
Nit: `501` seems to indicate the HTTP method (e.g. `PUT`, `POST`) is not supported. `504`, may be more semantically accurate. https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/503
```suggestion
        503,
```